src/system_extra.rs

```diff
@@ -0,0 +1,112 @@
+//! Extra system utilities.
+//!
+//! This modules contains an extension trait for the System trait which adds useful transformation
+//! functions.
+
+use ecs::prelude::{Read, System};
+use shred::{RunningTime, SystemData};
+
+/// Extra functionality associated systems.
+pub trait SystemExtra {
+    /// Make a system pausable by tying it to a specific value of a resource.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use amethyst::{
+    ///     ecs::{System, Write},
+    ///     shred::DispatcherBuilder,
+    ///     prelude::*,
+    /// };
+    ///
+    /// #[derive(PartialEq, Eq)]
+    /// enum State {
```

Comment:
the `State` name is confusing for pretty obvious reasons. Mind changing it? :)

Reply:
Not at all!